server/src/test/java/io/crate/execution/dml/upsert/TransportShardUpsertActionTest.java

```diff
@@ -158,6 +161,16 @@ public void prepare() throws Exception {
         when(indicesService.indexServiceSafe(charactersIndex)).thenReturn(indexService);
         when(indicesService.indexServiceSafe(partitionIndex)).thenReturn(indexService);
         indexShard = mock(IndexShard.class, Answers.RETURNS_MOCKS);
+        when(indexShard.index(any(Engine.Index.class))).thenAnswer(new Answer<>() {
+
+            @Override
+            public IndexResult answer(InvocationOnMock arg0) throws Throwable {
+                Engine.Index index = arg0.getArgument(0);
+                IndexResult indexResult = new IndexResult(1, index.primaryTerm(), index.seqNo(), true);
+                indexResult.setTranslogLocation(new Translog.Location(1, 1, 1));
+                return indexResult;
+            }
+        });
```

Comment:
How is this change related?

Reply:
`test_dynamic_insert_of_integer_upcasted_to_long_can_be_replicated` triggered the new assertions because the resultType didn't match, and it needs the correct seqNo + a translog location